components/config/src/config/languages.rs

```diff
@@ -30,6 +30,113 @@ pub struct LanguageOptions {
     pub translations: HashMap<String, String>,
 }
 
+impl LanguageOptions {
+    /// Merges self with another LanguageOptions, panicking if 2 equivalent fields are not None,
+    /// empty or of default value.
+    pub fn merge(&mut self, other: &LanguageOptions) {
+        match &self.title {
+            None => self.title = other.title.clone(),
+            Some(self_title) => match &other.title {
+                Some(other_title) => panic!(
+                    "Error: `title` for default language is specified twice, as {:?} and {:?}.",
+                    self_title, other_title
+                ),
+                None => (),
+            },
+        };
+
+        match &self.description {
+            None => self.description = other.description.clone(),
+            Some(self_description) => match &other.description {
+                Some(other_description) => panic!("Error: `description` for default language is specified twice, as {:?} and {:?}.", self_description, other_description),
+                None => (),
+            },
+        };
+
+        self.generate_feed = self.generate_feed || other.generate_feed;
+
+        match &self.feed_filename == "atom.xml" {
+            // "atom.xml" is default value.
+            true => self.feed_filename = other.feed_filename.clone(),
+            false => match &other.feed_filename.is_empty() {
+                false => panic!("Error: `feed filename` for default language is specifiec twice, as {:?} and {:?}.", self.feed_filename, other.feed_filename),
+                true => (),
+            },
+        };
+
+        match &self.taxonomies.is_empty() {
+            true => self.taxonomies = other.taxonomies.clone(),
+            false => match &other.taxonomies.is_empty() {
+                false => panic!("Error: `taxonomies` for default language is specifiec twice, as {:?} and {:?}.", self.taxonomies, other.taxonomies),
+                true => (),
+            },
+        };
+
+        self.build_search_index = self.build_search_index || other.build_search_index;
+
+        match self.search == search::Search::default() {
+            true => self.search = other.search.clone(),
+            false => match self.search == other.search {
+                true => (),
+                false => panic!(
+                    "Error: `search` for default language is specified twice, as {:?} and {:?}.",
+                    self.search, other.search
+                ),
+            },
+        };
+
+        match &self.translations.is_empty() {
+            true => self.translations = other.translations.clone(),
+            false => match &other.translations.is_empty() {
+                true => (),
+                false => panic!("Error: `translations` for default language is specified twice, as {:?} and {:?}.", self.translations, other.translations),
+            },
+        };
+    }
+
+    /// Checks whether self is of default value.
```

Comment:
You can just instantiate a default struct and compare them

Reply:
Don't know how that slipped my mind, or to compare the complete structs with each other instead of field by field.